corehq/apps/api/query_adapters.py

```diff
@@ -58,7 +64,22 @@ def count(self):
     def __getitem__(self, item):
         if isinstance(item, slice):
             limit = item.stop - item.start
-            result = GroupES().domain(self.domain).sort('name').size(limit).start(item.start).run()
-            return map(Group.wrap, result.hits)
+            result = GroupES().domain(self.domain).sort('name.exact').size(limit).start(item.start).run()
+            groups = [Group.wrap(group) for group in result.hits]
+
+            user_ids = {user_id for group in groups for user_id in group.users}
+            active_user_ids = (
+                UserES().domain(self.domain)
+                .user_ids(user_ids)
+                .is_active(True)
+                .values_list('_id', flat=True)
+            )
+
+            for group in groups:
+                group.__dict__['get_user_ids'] = lambda: [
```

Comment:
Yeah, I had trouble coming up with a good way to do this. A different option I considered was to set `group._precomputed_active_users` here and then have `get_user_ids` check for the existence of that first. Maybe that's better?

I sort of like your memoized idea since it is conceptually exactly what I'm intending, but the mechanics of that are I think not part of the memoized public interface.

I'm now noticing that what I did here is just actually incorrect; since I have this under a lambda, it will at the time of evaluation use the last value `group` had, and not the value `group` has at the time of the creation of the lambda.

I could just do `group.__dict__['get_user_ids'] = [...]` directly (since tastypie will take either a function or a value), but I'm now thinking my first idea was better, more explicit, cleaner—even though it does involve directly editing the `Group` class.

Open to other ideas, but that's currently my top one

Reply:
> A different option I considered was to set group._precomputed_active_users here and then have get_user_ids check for the existence of that first.

Yeah, I was thinking of that too.  I do like that option.

> I'm now noticing that what I did here is just actually incorrect; since I have this under a lambda, it will at the time of evaluation use the last value group had, and not the value group has at the time of the creation of the lambda.

Wow, I would not have guessed that.  Python variable scoping sometimes catches me way off guard.
```python
In [1]: class Group: 
   ...:     def __init__(self, n): 
   ...:         self.n = n 
   ...:                                                                                

In [2]: groups = [Group(1), Group(2), Group(3)]                                        

In [3]: for group in groups: 
   ...:     group.foo = lambda: group.n 
   ...:                                                                                

In [4]: [(g.n, g.foo()) for g in groups]                                               
Out[4]: [(1, 3), (2, 3), (3, 3)]
```

I do like the idea of editing the `Group` class for this.  Then at least if someone is banging their head on the desk trying to figure out how `Group.get_user_ids` is working, that will point them in the right direction.